mongorestore/restore.go

```diff
@@ -339,6 +340,11 @@ func fixDottedHashedIndex(index IndexDocument) {
 	}
 }
 
+type chunk struct {
```

Comment:
what's the rationale for using this `chunk` struct instead of just having a pool of `bson.Raw`s

Reply:
If I understand correctly, I think the reason is to insert the correct number of remaining documents in the last batch. We don't want to insert all 1000 array elements if only the first 50 were new elements. 

But, we can just take a new slice of the array instead. That would be a cleaner way of doing it and we could just use a pool of `[]bson.Raw`s.

Reply:
I agree that a pool of `[]bson.Raw` would be preferable to the `chunk` struct. At a higher level though, I'm wondering why we are batching at all. We weren't batching prior to this PR, and it doesn't seem that batching should be required to use a memory pool, as `pool.Get()` could return a `bson.Raw` instead of a `[]bson.Raw`.

I could imagine why batching might be beneficial (to reduce lock contention on `sync.Pool`, perhaps), but if we don't need to batch in order to get the perf improvements, I'd suggest we don't do it

Reply:
Ah I see what you're saying. I think doing batches is the correct thing to do. The reason is the `BufferedBulkInserter` buffers docs in batches before flushing. So we can't return documents to the pool one by one, we can only return batches of documents once the `BufferedBulkInserter` has flushed them. 

This does bring up an important point, the length of the `[]bson.Raw` in the pool should be `restore.OutputOptions.BulkBufferSize` to match the buffer size of the `BufferedBulkInserter`. Otherwise changing `--batchSize` would lead to errors.

Reply:
Makes sense -- SGTM!

Reply:
I think it was in order to reduce the number of object going through the `docChan` channel. It made a small difference for large number of documents (10M+ ? ) but I don't have the numbers anymore :/ I agree that it's a lot of extra complexity, so I switched the code back  `bson.Raw`